examples/fft_vs_geom_movie.py

```diff
@@ -0,0 +1,348 @@
+
+def ellip(mom):
+    """Convert moments dictionary into dictionary with ellipticity (e1, e2) and size (rsqr).
+    """
+    rsqr = mom['Ixx'] + mom['Iyy']
+    return dict(rsqr=rsqr, e1=(mom['Ixx']-mom['Iyy'])/rsqr, e2=2*mom['Ixy']/rsqr)
+
+def make_movie(args):
+    rng = galsim.BaseDeviate(args.seed)
+    u = galsim.UniformDeviate(rng)
+    # Generate 1D Gaussian random fields for each aberration.
+    t = np.arange(-args.n/2, args.n/2)
+    corr = np.exp(-0.5*t**2/args.ell**2)
+    pk = np.fft.fft(np.fft.fftshift(corr))
+    ak = np.sqrt(2*pk)
+    phi = np.random.uniform(size=(args.n, args.jmax))
+    zk = ak[:, None]*np.exp(2j*np.pi*phi)
+    aberrations = args.n/2*np.fft.ifft(zk, axis=0).real
+    measured_std = np.mean(np.std(aberrations, axis=0))
+    aberrations *= args.sigma/measured_std
+    aberrations -= np.mean(aberrations, axis=0)
+
+    # Generate an atmosphere.  Same procedure as in psf_wf_movie.py
+    Ellerbroek_alts = [0.0, 2.58, 5.16, 7.73, 12.89, 15.46]  # km
+    Ellerbroek_weights = [0.652, 0.172, 0.055, 0.025, 0.074, 0.022]
+    Ellerbroek_interp = galsim.LookupTable(Ellerbroek_alts, Ellerbroek_weights,
+                                           interpolant='linear')
+    alts = np.max(Ellerbroek_alts)*np.arange(args.nlayers)/(args.nlayers-1)
+    weights = Ellerbroek_interp(alts)  # interpolate the weights
+    weights /= sum(weights)  # and renormalize
+    max_speed = 20  # Pick (an arbitrary) maximum wind speed in m/s.
+    spd = []  # Wind speed in m/s
+    dirn = [] # Wind direction in radians
+    r0_500 = [] # Fried parameter in m at a wavelength of 500 nm.
+    for i in range(args.nlayers):
+        spd.append(u()*max_speed)  # Use a random speed between 0 and max_speed
+        dirn.append(u()*360*galsim.degrees)  # And an isotropically distributed wind direction.
+        r0_500.append(args.r0_500*weights[i]**(-3./5))
+        print ("Adding layer at altitude {:5.2f} km with velocity ({:5.2f}, {:5.2f}) m/s, "
+               "and r0_500 {:5.3f} m."
+               .format(alts[i], spd[i]*dirn[i].cos(), spd[i]*dirn[i].sin(), r0_500[i]))
+    if args.nlayers > 0:
+        atm = galsim.Atmosphere(r0_500=r0_500, speed=spd, direction=dirn, altitude=alts, rng=rng,
+                                time_step=args.time_step, screen_size=args.screen_size,
+                                screen_scale=args.screen_scale)
+    else:
+        atm = galsim.PhaseScreenList()
+
+    # Setup Fourier and geometric apertures
+    fft_aper = galsim.Aperture(args.diam, args.lam, obscuration=args.obscuration,
+                               pad_factor=args.pad_factor, oversampling=args.oversampling,
+                               nstruts=args.nstruts, strut_thick=args.strut_thick,
+                               strut_angle=args.strut_angle*galsim.degrees)
+    geom_aper = galsim.Aperture(args.diam, args.lam, obscuration=args.obscuration,
+                                pad_factor=args.geom_oversampling, oversampling=0.5,
```

Comment:
Good question, and worth elaborating somewhere, maybe in the docstrings for `PhaseScreenPSF` and `OpticalPSF`?

The reason is that `oversample=0.5` produces a (square) pupil plane array that precisely circumscribes the (circular) aperture.  (`oversample=1.0` produces a square twice this size, which is just large enough for two copies of the aperture to cleanly "slide" past each with periodic boundary conditions, which is what you need for no aliasing in Fourier optics.)  Setting `oversample` larger than 0.5 adds points to the pupil array that are never sampled when collecting phase gradients, and setting `oversample` smaller starts to clip pieces of the aperture that we do want to sample.